### PR TITLE
Convert package URIs to use top-level drake manifest

### PR DESCRIPTION
**Pull request by cottsay**
This change converts many manifest-relative file references to use the top-level `drake` manifest (added in #15468) and deprecates the individual manifests that are no longer used (using #15896).

The following manifests are being deprecated by this PR:
- [examples/atlas/package.xml](https://github.com/RobotLocomotion/drake/blob/master/examples/atlas/package.xml)
- [examples/manipulation_station/models/package.xml](https://github.com/RobotLocomotion/drake/blob/master/examples/manipulation_station/models/package.xml)
- [manipulation/models/allegro_hand_description/package.xml](https://github.com/RobotLocomotion/drake/blob/master/manipulation/models/allegro_hand_description/package.xml)
- [manipulation/models/franka_description/package.xml](https://github.com/RobotLocomotion/drake/blob/master/manipulation/models/franka_description/package.xml)
- [manipulation/models/iiwa_description/package.xml](https://github.com/RobotLocomotion/drake/blob/master/manipulation/models/iiwa_description/package.xml)
- [manipulation/models/jaco_description/package.xml](https://github.com/RobotLocomotion/drake/blob/master/manipulation/models/jaco_description/package.xml)
- [manipulation/models/wsg_50_description/package.xml](https://github.com/RobotLocomotion/drake/blob/master/manipulation/models/wsg_50_description/package.xml)
- [manipulation/models/ycb/package.xml](https://github.com/RobotLocomotion/drake/blob/master/manipulation/models/ycb/package.xml)
- ~~[multibody/parsing/test/process_model_directives_test/package.xml](https://github.com/RobotLocomotion/drake/blob/master/multibody/parsing/test/process_model_directives_test/package.xml)~~

Of those, I'd specifically like feedback on whether this is the right thing to do for the following manifests, which I'm concerned may be propagated to published ROS packages:
- [manipulation/models/franka_description/package.xml](https://github.com/RobotLocomotion/drake/blob/master/manipulation/models/franka_description/package.xml)
- [manipulation/models/iiwa_description/package.xml](https://github.com/RobotLocomotion/drake/blob/master/manipulation/models/iiwa_description/package.xml)
- [manipulation/models/jaco_description/package.xml](https://github.com/RobotLocomotion/drake/blob/master/manipulation/models/jaco_description/package.xml)

These are the remaining manifests which are not being deprecated by this PR:
- [multibody/parsing/test/box_package/package.xml](https://github.com/RobotLocomotion/drake/blob/master/multibody/parsing/test/box_package/package.xml)
- [multibody/parsing/test/package_map_test_package_conflicting/package.xml](https://github.com/RobotLocomotion/drake/blob/master/multibody/parsing/test/package_map_test_package_conflicting/package.xml)
- [multibody/parsing/test/package_map_test_packages/package_map_test_package_a/package.xml](https://github.com/RobotLocomotion/drake/blob/master/multibody/parsing/test/package_map_test_packages/package_map_test_package_a/package.xml)
- [multibody/parsing/test/package_map_test_packages/package_map_test_package_b/package.xml](https://github.com/RobotLocomotion/drake/blob/master/multibody/parsing/test/package_map_test_packages/package_map_test_package_b/package.xml)
- [multibody/parsing/test/package_map_test_packages/package_map_test_package_set/package_map_test_package_c/package.xml](https://github.com/RobotLocomotion/drake/blob/master/multibody/parsing/test/package_map_test_packages/package_map_test_package_set/package_map_test_package_c/package.xml)
- [multibody/parsing/test/package_map_test_packages/package_map_test_package_set/package_map_test_package_d/package.xml](https://github.com/RobotLocomotion/drake/blob/master/multibody/parsing/test/package_map_test_packages/package_map_test_package_set/package_map_test_package_d/package.xml)
- **[multibody/parsing/test/process_model_directives_test/package.xml](https://github.com/RobotLocomotion/drake/blob/master/multibody/parsing/test/process_model_directives_test/package.xml)**
- [multibody/parsing/test/sdf_parser_test/interface_api_test/package.xml](https://github.com/RobotLocomotion/drake/blob/master/multibody/parsing/test/sdf_parser_test/interface_api_test/package.xml)
- [multibody/parsing/test/sdf_parser_test/simple_robot1/package.xml](https://github.com/RobotLocomotion/drake/blob/master/multibody/parsing/test/sdf_parser_test/simple_robot1/package.xml)
- [multibody/parsing/test/sdf_parser_test/simple_robot2/package.xml](https://github.com/RobotLocomotion/drake/blob/master/multibody/parsing/test/sdf_parser_test/simple_robot2/package.xml)

For the most part, the remaining manifests seem to directly test the parsing and package mapping functionality in Drake and I'm not sure it would be appropriate to convert them, but I may be wrong in some of these cases. If you see any manifests that you believe should be deprecated (or vice-versa), please point them out.

I verified that all of the package URIs which properly resolved to files prior to this PR resolved properly after changing to the top-level manifest.

Following convention, I noted a removal date of ~~February~~ **March** 1st 2022 for the deprecated manifests.

Towards #10531

---
This change is [Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/15947)

### CODE REVIEW

**Review by cottsay — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/15947)* status: 8 unresolved discussions, LGTM missing from assignee jwnimmer-tri(platform), needs at least two assigned reviewers, commits need curation (https://drake.mit.edu/reviewable.html#curated-commits) (waiting on @cottsay and @jwnimmer-tri)

___
*[multibody/parsing/test/process_model_directives_test.cc, line 39 at r1](https://reviewable.io/reviews/robotlocomotion/drake/15947#-Mmy7EJG2hGhUuW6Qrar:-MnWmok70ckWYcESVyRZ:bxfgvu6) ([raw file](https://github.com/robotlocomotion/drake/blob/fd3733d40f67174cb17688a16b4ea13dd6c2e9f5/multibody/parsing/test/process_model_directives_test.cc#L39)):*
<details><summary><i>Previously, jwnimmer-tri (Jeremy Nimmer) wrote…</i></summary><blockquote>

With this change, we've lost the functional test coverage that a custom PackageMap provided by the user will be obeyed by the model directives.

There are probably ways to add back coverage of that even with just the `drake` package, but the simplest answer is probably just to revert the deprecation of this test's custom package.xml.
</blockquote></details>

I was basing the removal of this manifest on this comment, which seems to suggest that it should be removed once the top-level manifest became available: https://github.com/RobotLocomotion/drake/blob/dadf4e37086da47f857068b99c06a24b22d29b0f/multibody/parsing/test/process_model_directives_test/simple_model.sdf#L20-L21

If you think that is not correct, then the comment should be updated.
___
*[examples/atlas/sdf/cinder_block_2/model.sdf, line 26 at r1](https://reviewable.io/reviews/robotlocomotion/drake/15947#-Mmx6bSa5k8kEeutK6uY:-MnWn9Ag-_tKwcjmxGFL:b-q0eb3y) ([raw file](https://github.com/robotlocomotion/drake/blob/1345a22cefe721abb4235d04d39e6ca9cb1ff3a3/examples/atlas/sdf/cinder_block_2/model.sdf#L26)):*
<details><summary><i>Previously, jwnimmer-tri (Jeremy Nimmer) wrote…</i></summary><blockquote>

The code being edited here is commented out.  I don't see any reason to make any change here?  The `model.config` file is not being changed in this PR.
</blockquote></details>

I updated the path because even though it's commented out, it still refers to a file present in the tree. That said, you're right, it has been commented out.

**Review by cottsay — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/15947)* status: 4 unresolved discussions, LGTM missing from assignee jwnimmer-tri(platform), needs at least two assigned reviewers, commits need curation (https://drake.mit.edu/reviewable.html#curated-commits) (waiting on @cottsay and @jwnimmer-tri)

___
*[examples/atlas/package.xml, line 14 at r2](https://reviewable.io/reviews/robotlocomotion/drake/15947#-MnWqmYI63vseze_XljR:-MnWsr_ZD_IHipKob7Oy:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/3da890ba8b3e3f8e834d04d6a6d8d559af1271e0/examples/atlas/package.xml#L14)):*
<details><summary><i>Previously, jwnimmer-tri (Jeremy Nimmer) wrote…</i></summary><blockquote>

We need 2022-03-01 throughout now, since it will several days into November by the time this PR merges to master.
</blockquote></details>

Done.
___
*[examples/atlas/sdf/cinder_block_2/model.sdf, line 26 at r1](https://reviewable.io/reviews/robotlocomotion/drake/15947#-Mmx6bSa5k8kEeutK6uY:-MnWsKGK-o2UCc-HChgw:b-7aitaz) ([raw file](https://github.com/robotlocomotion/drake/blob/1345a22cefe721abb4235d04d39e6ca9cb1ff3a3/examples/atlas/sdf/cinder_block_2/model.sdf#L26)):*
<details><summary><i>Previously, jwnimmer-tri (Jeremy Nimmer) wrote…</i></summary><blockquote>

OK Yes, but there was no `cinder_block_2` package that we deprecated.  I as understand it, this sdf is using `model.config` in the current directory, which remains unchanged.  (So either, it will still remain working, or will be equally as broken as previously.  In any case, orthogonal to the atlas package.xml changes.)  I should have emphasized the "model.config isn't changing"; the fact that it's commented out was mostly irrelevant.
</blockquote></details>

Thanks for clarifying.

**Review by jwnimmer-tri — COMMENTED** [review]
> Reviewed 1 of 14 files at r2, 8 of 8 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/15947)* status: 3 unresolved discussions, LGTM missing from assignee jwnimmer-tri(platform), needs at least two assigned reviewers, commits need curation (https://drake.mit.edu/reviewable.html#curated-commits) (waiting on @cottsay and @jwnimmer-tri)

___
*[multibody/parsing/test/process_model_directives_test.cc, line 39 at r1](https://reviewable.io/reviews/robotlocomotion/drake/15947#-Mmy7EJG2hGhUuW6Qrar:-MnWt__k9Rphd8ZenIZu:b-688e07) ([raw file](https://github.com/robotlocomotion/drake/blob/fd3733d40f67174cb17688a16b4ea13dd6c2e9f5/multibody/parsing/test/process_model_directives_test.cc#L39)):*
<details><summary><i>Previously, cottsay (Scott K Logan) wrote…</i></summary><blockquote>

I was basing the removal of this manifest on this comment, which seems to suggest that it should be removed once the top-level manifest became available: https://github.com/RobotLocomotion/drake/blob/dadf4e37086da47f857068b99c06a24b22d29b0f/multibody/parsing/test/process_model_directives_test/simple_model.sdf#L20-L21

If you think that is not correct, then the comment should be updated.
</blockquote></details>

If the thesis is that `simple_model.sdf` should use the `package://drake` because its purpose is to be a simple example target for the yaml syntax, and _not_ to be a test of the "custom Parser" feature, then the TODO was accurate and the proposed change to `simple_model.sdf` is correct.

However, that leaves the "Users can supply a custom Parser argument which we will obey" logic untested:

https://github.com/RobotLocomotion/drake/blob/4d6b7bfd134f31447def5627d10f43f31f870ea9/multibody/parsing/process_model_directives.cc#L249

In other words, if the implementation on that line ignored the optional `parser` argument and always just created a new `Parser`, then that new parser would automatically have the `package://drake` pre-populated, and the test would still pass, because it only ever used resources from within `package://drake`.  We need to have a unit test that fails when the `parser` argument is ignored.

I don't have an opinion about whether we achieve that by adding a _new_ test case for that, or by leaving the simple_model ones alone.

**Review by cottsay — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/15947)* status: 3 unresolved discussions, LGTM missing from assignee jwnimmer-tri(platform), needs at least two assigned reviewers, commits need curation (https://drake.mit.edu/reviewable.html#curated-commits) (waiting on @jwnimmer-tri)

___
*[multibody/parsing/test/process_model_directives_test.cc, line 39 at r1](https://reviewable.io/reviews/robotlocomotion/drake/15947#-Mmy7EJG2hGhUuW6Qrar:-MnXDlXP4gbc17gC0oPc:bd09to) ([raw file](https://github.com/robotlocomotion/drake/blob/fd3733d40f67174cb17688a16b4ea13dd6c2e9f5/multibody/parsing/test/process_model_directives_test.cc#L39)):*
<details><summary><i>Previously, jwnimmer-tri (Jeremy Nimmer) wrote…</i></summary><blockquote>

If the thesis is that `simple_model.sdf` should use the `package://drake` because its purpose is to be a simple example target for the yaml syntax, and _not_ to be a test of the "custom Parser" feature, then the TODO was accurate and the proposed change to `simple_model.sdf` is correct.

However, that leaves the "Users can supply a custom Parser argument which we will obey" logic untested:

https://github.com/RobotLocomotion/drake/blob/4d6b7bfd134f31447def5627d10f43f31f870ea9/multibody/parsing/process_model_directives.cc#L249

In other words, if the implementation on that line ignored the optional `parser` argument and always just created a new `Parser`, then that new parser would automatically have the `package://drake` pre-populated, and the test would still pass, because it only ever used resources from within `package://drake`.  We need to have a unit test that fails when the `parser` argument is ignored.

I don't have an opinion about whether we achieve that by adding a _new_ test case for that, or by leaving the simple_model ones alone.
</blockquote></details>

Thanks for clarifying.

I tried to implement a new test specifically for this behavior, but it got very contrived and I'm not sure it provided all that much long-term value. At this point, I believe that this manifest falls under the "directly test the parsing and package mapping functionality in Drake" category I mentioned in the PR summary, and should be left alone.
___
*[multibody/parsing/test/process_model_directives_test/package.xml, line 15 at r1](https://reviewable.io/reviews/robotlocomotion/drake/15947#-Mmy1MRB-n6M5BAUb16B:-MnXEaaE2VosfxrRiAnr:b7m24bn) ([raw file](https://github.com/robotlocomotion/drake/blob/1345a22cefe721abb4235d04d39e6ca9cb1ff3a3/multibody/parsing/test/process_model_directives_test/package.xml#L15)):*
<details><summary><i>Previously, jwnimmer-tri (Jeremy Nimmer) wrote…</i></summary><blockquote>

After deprecation, we should just make this data private to Drake.  There is no reason that downstream code should be able rely on a stable API for our test data.  As such, the "Please use" advice here should change (or be removed).

... but see also the discussion in the unit test -- I'm not even sure that we want to deprecate here at all.
</blockquote></details>

Per the other thread, I think this manifest should be left alone.

**Review by jwnimmer-tri — COMMENTED** [review]
> Reviewed 5 of 62 files at r1, 7 of 7 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/15947)* status: 2 unresolved discussions, LGTM missing from assignee jwnimmer-tri(platform), needs at least two assigned reviewers, commits need curation (https://drake.mit.edu/reviewable.html#curated-commits) (waiting on @cottsay and @jwnimmer-tri)

*[a discussion](https://reviewable.io/reviews/robotlocomotion/drake/15947#-Mmx83Tg7uD0DtWrS31O:-MnXdKsa4c0zQcZsSRxT:bexhb9e) (no related file):*
<details><summary><i>Previously, jwnimmer-tri (Jeremy Nimmer) wrote…</i></summary><blockquote>

Working

This is ready to test now.  I'll work on that.
</blockquote></details>

Tested:
* iiwa14 no collision
* jaco hand sphere collision
* jaco arm sphere collision
* franka panda arm -- meshcat has no textures? investigating.
* manipulation station ee teleop sliders, with wsg

---
___
*[multibody/parsing/test/process_model_directives_test/simple_model.sdf, line 20 at r4](https://reviewable.io/reviews/robotlocomotion/drake/15947#-MnXcBot95Sz8GaRrahE:-MnXcBot95Sz8GaRrahF:bqh3lux) ([raw file](https://github.com/robotlocomotion/drake/blob/f3a2272f2897ac087e29e29649ea2c192a794fec/multibody/parsing/test/process_model_directives_test/simple_model.sdf#L20)):*
> ```
>           -->
>           <drake:diffuse_map>
>             
> ```

As I understand it, we both think this TODO should be removed now -- there is no further action required?

**Review by jwnimmer-tri — COMMENTED** [review]
:lgtm: feature.

+@EricCousineau-TRI for platform review per (next) scheduled, please.

> *[Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/15947#-:-MnXhsIY8S516tBo9L9K:buuvtl1)* status: 1 unresolved discussion, LGTM missing from assignee EricCousineau-TRI(platform), commits need curation (https://drake.mit.edu/reviewable.html#curated-commits) (waiting on @cottsay and @EricCousineau-TRI)

*[a discussion](https://reviewable.io/reviews/robotlocomotion/drake/15947#-Mmx83Tg7uD0DtWrS31O:-MnXhn_BE50WB-jkZA0Z:b-8ab0mj) (no related file):*
> franka panda arm -- meshcat has no textures?

Meh.  Same thing on master, so not a regression here.

---

**Review by cottsay — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/15947)* status: 1 unresolved discussion, LGTM missing from assignee EricCousineau-TRI(platform), commits need curation (https://drake.mit.edu/reviewable.html#curated-commits) (waiting on @EricCousineau-TRI and @jwnimmer-tri)

___
*[multibody/parsing/test/process_model_directives_test/simple_model.sdf, line 20 at r4](https://reviewable.io/reviews/robotlocomotion/drake/15947#-MnXcBot95Sz8GaRrahE:-MnY1-SQ3013cJAt6h39:bu4ephv) ([raw file](https://github.com/robotlocomotion/drake/blob/f3a2272f2897ac087e29e29649ea2c192a794fec/multibody/parsing/test/process_model_directives_test/simple_model.sdf#L20)):*
<details><summary><i>Previously, jwnimmer-tri (Jeremy Nimmer) wrote…</i></summary><blockquote>

As I understand it, we both think this TODO should be removed now -- there is no further action required?
</blockquote></details>

Dropped.

**Review by jwnimmer-tri — COMMENTED** [review]
> Reviewed 1 of 1 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/15947)* status: LGTM missing from assignee EricCousineau-TRI(platform), commits need curation (https://drake.mit.edu/reviewable.html#curated-commits) (waiting on @EricCousineau-TRI)

**Review by EricCousineau-TRI — COMMENTED** [review]
:lgtm_strong: platform

> Reviewed 45 of 62 files at r1, 2 of 14 files at r2, 7 of 8 files at r3, 1 of 1 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/15947#-:-MnfVus468NmicMNGfqq:b-71a3kq)* status: commits need curation (https://drake.mit.edu/reviewable.html#curated-commits) (waiting on @cottsay)

*[a discussion](https://reviewable.io/reviews/robotlocomotion/drake/15947#-Mmy4Hxa15XcEU1Kb5qv:-MnfVtgp4-IBUB8Snpbm:b-w196av) (no related file):*
<details><summary><i>Previously, jwnimmer-tri (Jeremy Nimmer) wrote…</i></summary><blockquote>

OK the Anzu patch is sufficiently ready.
</blockquote></details>

OK Thanks!

---

**Review by EricCousineau-TRI — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/15947)* status: commits need curation (https://drake.mit.edu/reviewable.html#curated-commits) (waiting on @cottsay)

*[a discussion](https://reviewable.io/reviews/robotlocomotion/drake/15947#-Mmy4Hxa15XcEU1Kb5qv:-MnfXE1e1xHIJ9I3YtLa:b6742v9) (no related file):*
<details><summary><i>Previously, EricCousineau-TRI (Eric Cousineau) wrote…</i></summary><blockquote>

OK Thanks!
</blockquote></details>

OK Confirmed in Anzu PR 7913

---
